### DISCUSSION TITLE
关于高级篇-Zero-Shot Prompts的一点疑惑

### DISCUSSION THREAD

**Discussion by etc1208** [thinkingjimmy/Learning-Prompt#47] [Q&A] [answered]
[原文](https://learningprompt.wiki/docs/tutorial-extras/Zero-Shot%20Prompts)
其中提到：“首先各位要清楚像 ChatGPT 这类产品，它是一个统计语言模型，本质上是基于过去看到过的所有数据，用统计学意义上的预测结果进行下一步的输出（这也就是为什么你在使用 ChatGPT 的时候，它的答案是一个字一个字地吐出来，而不是直接给你的原因，因为答案是一个字一个字算出来的”

“答案是一个字一个字地吐出来” 不是因为使用了流式输出的结果吗？

**Comment by thinkingjimmy** [comment]
不确定你说的流式输出是不是我理解的流式输出 😂

首先这个一个字一个字吐出来不是前端刻意实现的，而是模型（或者说后端）返回结果的时候，就是一个个字算出来，然后一个字一个字吐出来的。

举个例子：
当你问 AI 一个问题时，后端会先算出答案的第一个字是 A，然后再算出来第二个字是 B，接着是 C....
最后组成一句话。

**Comment by KonghaYao** [comment, marked as answer]
GPT 原理上是根据上一文猜测下一个字符，所以确实是一个一个推断出的。生成文本这个行为像是水流从后端源头流向了前端，我们前端也是用流式接收并不断更新这个结果的。

同时，由于程序功能上的需要，肯定是把最新的结果告诉前端是最好的，不然等后端模型一句话要五六秒钟，用户失去兴趣的可能很大。

但这个并不是服务器后端强制 **使用了流式输出的结果**，这个形式是 **模型的生成方式和程序、需求上的设计决定的**。

**Reply by thinkingjimmy** [reply]
很清晰的回答 👍
